evalml/tests/component_tests/test_estimators.py

```diff
@@ -85,3 +87,91 @@ def test_all_estimators_check_fit_input_type_regression(data_type, X_y_regressio
         component = helper_functions.safe_init_component_with_njobs_1(component_class)
         component.fit(X, y)
         component.predict(X)
+
+
+def test_estimator_predict_output_type(X_y_binary, helper_functions):
+    X_np, y_np = X_y_binary
+    assert isinstance(X_np, np.ndarray)
+    assert isinstance(y_np, np.ndarray)
+    y_list = list(y_np)
+    X_df_no_col_names = pd.DataFrame(X_np)
+    range_index = pd.RangeIndex(start=0, stop=X_np.shape[1], step=1)
+    X_df_with_col_names = pd.DataFrame(X_np, columns=['x' + str(i) for i in range(X_np.shape[1])])
+    y_series_no_name = pd.Series(y_np)
+    y_series_with_name = pd.Series(y_np, name='target')
+    datatype_combos = [(X_np, y_np, range_index, np.unique(y_np)),
+                       (X_np, y_list, range_index, np.unique(y_np)),
+                       (X_df_no_col_names, y_series_no_name, range_index, y_series_no_name.unique()),
+                       (X_df_with_col_names, y_series_with_name, X_df_with_col_names.columns, y_series_with_name.unique())]
+
+    for component_class in _all_estimators_used_in_search():
+        for X, y, X_cols_expected, y_cols_expected in datatype_combos:
+            print('Checking output of predict for estimator "{}" on X type {} cols {}, y type {} name {}'
```

Comment:
Do we need this print statement now?

Reply:
Oo I didn't add this, I just moved this--I believe @dsherry added this a while ago because it helped when testing. I think we should keep it--I also noticed it helped me to see which estimator was erroring out while I was debugging 😁